src/properties/text_property.rs

```diff
@@ -1,292 +1,1043 @@
+use std::hash::{Hash, Hasher};
 use std::{
     fmt::Debug,
     io::{Cursor, Read, Seek, Write},
 };
 
 use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use indexmap::IndexMap;
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+use ordered_float::OrderedFloat;
+use unreal_helpers::{UnrealReadExt, UnrealWriteExt};
 
+use crate::custom_version::FEditorObjectVersion;
+use crate::properties::int_property::UInt64Property;
+use crate::properties::struct_types::DateTime;
 use crate::{
     cursor_ext::{ReadExt, WriteExt},
     error::{DeserializeError, Error},
 };
 
-use super::{impl_write, PropertyOptions, PropertyTrait};
+use super::{impl_read, impl_read_header, impl_write, PropertyOptions, PropertyTrait};
 
 /// A property that stores GVAS Text.
-#[derive(Clone, PartialEq, Eq, Hash)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
-pub enum TextProperty {
-    /// An empty `TextProperty`.
-    Empty(
-        // Workaround for https://github.com/serde-rs/json/issues/664
-        [u8; 0],
-    ),
-    /// A rich `TextProperty`.
-    Rich(RichText),
-    /// A simple `TextProperty`.
-    Simple(Vec<String>),
-    /// `TextProperty` type 8.
-    Type8(String, String, String),
+pub struct TextProperty {
+    /// Value
+    pub value: FText,
```

Comment:
Since the gvas2json utility exposes the internal data structure of this library, I'd prefer to keep the data as flat as possible, ie:

```suggestion
    /// Text flags
    pub flags: u32,
    /// Text history
    pub history: FTextHistory,
```

Reply:
Wouldn't serde(flatten) fix this?

Reply:
Ah yes, I forgot about `flatten`. That works for me!